e2e/internal/kubeclient/deploy.go

```diff
@@ -22,6 +22,27 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
+// WaitCondition is an enum type for the possible wait conditions when using `kubeclient.WaitFor`.
+type WaitCondition int
+
+const (
+	_ WaitCondition = iota
+	// Ready waits until the resource becomes ready.
+	Ready
+	// Added waits until a `watch.Added` event occurs.
+	Added
+	// Modified waits until a `watch.Modified` event occurs.
+	Modified
+	// Deleted waits until a `watch.Deleted` event occurs.
+	Deleted
+	// Bookmark waits until a `watch.Bookmark` event occurs.
+	Bookmark
+	// Running waits until all containers of all pods of the resource are running.
+	Running
+	// InitContainersRunning waits until all initial containers of all pods of the resource are running.
+	InitContainersRunning
```

Comment:
I don't know of any plans to use the others right now, I just introduced them because the functionality basically was built in already and needed nearly no extra code. If you think that, for the sake of clarity, this should be removed, I'm fine with that :)